sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala

```diff
@@ -3097,4 +3097,19 @@ class JDBCV2Suite extends QueryTest with SharedSparkSession with ExplainSuiteHel
     assert(rows.contains(Row(null)))
     assert(rows.contains(Row("a a a")))
   }
+
+  test("SPARK-50792 Format binary data as a binary literal in JDBC.") {
```

Comment:
I think for dialect-specific fixes, it's sufficient to put the tests `V2JDBCTest.scala`. This test suite is for testing the shared code paths for all dialects.

Reply:
Right, this PR is somewhat complicated, it has changes in shared code paths to support `Binary Literal`. And also has changes in MsSqlServerDialect, OracleDialect, and PostgresDialect. If this test is not necessary, we can remove it.